x-pack/plugins/reporting/server/export_types/png_v2/png_v2.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import apm from 'elastic-apm-node';
+import { LicenseType } from '@kbn/licensing-plugin/server';
+import { CancellationToken, TaskRunResult } from '@kbn/reporting-common';
+import { Writable } from 'stream';
+import * as Rx from 'rxjs';
+import { finalize, map, mergeMap, takeUntil, tap } from 'rxjs';
+import { SerializableRecord } from '@kbn/utility-types';
+import { LocatorParams } from '../../../common';
+import {
+  LICENSE_TYPE_CLOUD_STANDARD,
+  LICENSE_TYPE_ENTERPRISE,
+  LICENSE_TYPE_GOLD,
+  LICENSE_TYPE_PLATINUM,
+  LICENSE_TYPE_TRIAL,
+  PNG_JOB_TYPE_V2,
+  PNG_REPORT_TYPE_V2,
+  REPORTING_TRANSACTION_TYPE,
+} from '../../../common/constants';
+import { decryptJobHeaders, ExportType, generatePngObservable } from '../common';
+import { JobParamsPNGV2, TaskPayloadPNGV2 } from './types';
+import { getFullRedirectAppUrl } from '../common/v2/get_full_redirect_app_url';
+
+/*
+ * @TODO move to be within @kbn/reporting-export-types
+ */
+
+export class PngExportType extends ExportType<JobParamsPNGV2, TaskPayloadPNGV2> {
+  id = PNG_REPORT_TYPE_V2;
+  name = 'PNG';
+  jobType = PNG_JOB_TYPE_V2;
+  jobContentEncoding = 'base64' as const;
+  jobContentExtension = 'png' as const;
+  validLicenses: LicenseType[] = [
+    LICENSE_TYPE_TRIAL,
+    LICENSE_TYPE_CLOUD_STANDARD,
+    LICENSE_TYPE_GOLD,
+    LICENSE_TYPE_PLATINUM,
+    LICENSE_TYPE_ENTERPRISE,
+  ];
+
+  constructor(...args: ConstructorParameters<typeof ExportType>) {
+    super(...args);
+    const logger = args[2];
+    this.logger = logger.get('png-export-v2');
```

Comment:
A `logger` field is added to the class instance after the `super()` call is through. That means, it looks like this could be simplified (for all the export type declarations):
```suggestion
    super(...args);
    this.logger = this.logger.get('png-export-v2');
```